app/src/main/java/ch/epfl/sdp/appart/user/UserProfileActivity.java

```diff
@@ -0,0 +1,198 @@
+package ch.epfl.sdp.appart.user;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.res.ResourcesCompat;
+import ch.epfl.sdp.appart.R;
+import ch.epfl.sdp.appart.database.Database;
+import ch.epfl.sdp.appart.database.FirebaseDB;
+import android.graphics.drawable.Drawable;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * This class manages the UI of the user profile and calls
+ * firebase class in order to manage the updated information.
+ */
+public class UserProfileActivity extends AppCompatActivity {
+
+    /* temporary user */
+    private User sessionUser;
+
+    /* Firestore database instance for db updates */
+    private Database db;
```

Comment:
We should use the hilt injection

Reply:
Indeed, thanks! I just committed with this adjustment.